### ISSUE TITLE
com.google.gson.stream.MalformedJsonException while trying to run test

### ISSUE THREAD

**Issue by YuvalItzchakov** [pantsbuild/intellij-pants-plugin#383]
IntelliJ IDEA 2018.3.1 (Ultimate Edition)
Build #IU-183.4588.61, built on December 4, 2018
JRE: 1.8.0_152-release-1343-b16 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.1

Project type: Scala

While writing a test and attempting to run it using CTRL+SHIFT+R shortcut, I get a prompt from IDEA indicating there was an error running the command:

```
2018-12-12 11:57:44,654 [  68730]  ERROR - penapi.actionSystem.impl.Utils - IntelliJ IDEA 2018.3.1  Build #IU-183.4588.61 
2018-12-12 11:57:44,654 [  68730]  ERROR - penapi.actionSystem.impl.Utils - JDK: 1.8.0_152-release; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o 
2018-12-12 11:57:44,654 [  68730]  ERROR - penapi.actionSystem.impl.Utils - OS: Mac OS X 
2018-12-12 11:57:44,655 [  68731]  ERROR - penapi.actionSystem.impl.Utils - Plugin to blame: Pants Support version: 1.11.0 
2018-12-12 11:57:44,655 [  68731]  ERROR - penapi.actionSystem.impl.Utils - Last Action: EditorCopy 
2018-12-12 11:57:44,662 [  68738]  ERROR - penapi.actionSystem.impl.Utils - update failed for AnAction(com.intellij.execution.actions.CreateAction) with ID=CreateRunConfiguration 
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Unterminated array at line 1 column 13 path $[2]
	at com.google.gson.Gson.fromJson(Gson.java:942)
	at com.google.gson.Gson.fromJson(Gson.java:892)
	at com.google.gson.Gson.fromJson(Gson.java:841)
	at com.twitter.intellij.pants.util.PantsUtil.hydrateTargetAddresses(PantsUtil.java:798)
	at com.twitter.intellij.pants.util.PantsUtil.getTargetAddressesFromModule(PantsUtil.java:446)
	at com.twitter.intellij.pants.util.PantsUtil.getNonGenTargetAddresses(PantsUtil.java:459)
	at com.twitter.intellij.pants.execution.PantsConfigurationContext.validatesAndCreate(PantsConfigurationContext.java:43)
	at com.twitter.intellij.pants.execution.PantsJUnitTestRunConfigurationProducer.setupConfigurationFromContext(PantsJUnitTestRunConfigurationProducer.java:38)
	at com.twitter.intellij.pants.execution.PantsJUnitTestRunConfigurationProducer.setupConfigurationFromContext(PantsJUnitTestRunConfigurationProducer.java:31)
	at com.intellij.execution.actions.RunConfigurationProducer.createConfigurationFromContext(RunConfigurationProducer.java:79)
	at com.intellij.execution.actions.RunConfigurationProducer.findOrCreateConfigurationFromContext(RunConfigurationProducer.java:167)
	at com.intellij.execution.actions.PreferredProducerFind.doGetConfigurationsFromContext(PreferredProducerFind.java:108)
	at com.intellij.execution.actions.PreferredProducerFind.getConfigurationsFromContext(PreferredProducerFind.java:92)
	at com.intellij.execution.actions.PreferredProducerFind.findConfigurationFromContext(PreferredProducerFind.java:133)
	at com.intellij.execution.actions.PreferredProducerFind.createConfiguration(PreferredProducerFind.java:31)
	at com.intellij.execution.actions.ConfigurationContext.createConfiguration(ConfigurationContext.java:147)
	at com.intellij.execution.actions.ConfigurationContext.getConfiguration(ConfigurationContext.java:138)
	at com.intellij.execution.actions.BaseRunConfigurationAction.update(BaseRunConfigurationAction.java:193)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:171)
	at com.intellij.openapi.actionSystem.impl.Utils.doUpdate(Utils.java:202)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:151)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:295)
```

I tried searching for the configuration JSON which is causing this error with no success.

Running Pants Refresh resolves the issue

**Comment by wisechengyi** [comment]
You should be able to find the json output by looking at `./pants server --open` and look for the `export` command with `--output-file=....`

**Comment by DJFliX** [comment]
@wisechengyi I followed your instructions and found some massive JSON output file (256kb) that got validated by jsonlint.com successfully. Contrary to OP I am not able to 'fix' my IDE by refreshing pants. Sometimes I'm able to debug a test suite (because running it usually doesn't even work at all). Once a Pants operation (test or debug test) fails the option is greyed out in they IDE and stays that way. Is there any way I can help find possible causes? The error message mentions a path (`at line 1 column 12 path $[2]`) but I'm not sure where to evaluate it in the JSON.

**Comment by wisechengyi** [comment]
What's your pants version?

**Comment by DJFliX** [comment]
> What's your pants version?

```|bash
➜  codebase git:(develop) ✗ pants -v          
1.9.0
```

**Comment by wisechengyi** [comment]
The issue seems specific to your env I reckon. https://stackoverflow.com/questions/11484353/gson-throws-malformedjsonexception talks about some causes but I'm not sure how to debug it in your case.

Can you try loading the project in a more sanitized environment, e.g. a VM?

**Comment by DJFliX** [comment]
I will try on Monday. My colleague gets the exact same error with the same pants IDEA version on his MacBook. I think it makes more sense to create a small pants test project and test that on both our machines as I'm pretty sure that loading our main project in a VM will yield the exact same results.

**Comment by DJFliX** [comment]
I tried loading a hello world project with a test and that loaded/ran just fine in IntelliJ. Running pants commands from the terminal also works fine (even on the project that breaks the pants plugin in IntelliJ). Could it have something to do with the structure go our BUILD files/pants.ini file? Or a pants plugin that is interfering with the intellij-pants-plugin operations?

**Comment by wisechengyi** [comment]
Is it possible that the project dirs/paths could contain non-canonical english characters?

**Comment by DJFliX** [comment]
@wisechengyi Thanks for looking into it and spending some time. Much appreciated!
> Is it possible that the project dirs/paths could contain uncanonical non-english characters?
This was not the case. 

@Vakha proposed to clone the repo into a new directory. Instead I decided to nuke all pants folders (`rm -rf .idea/pants-projects .pants.d .pants.workdir.file_lock .pids dist .cache`) and re-import the project in IntelliJ. Probably only removing the `.idea/pants-projects` folder would have been sufficient to fix the issue.

**Comment by DJFliX** [comment]
@wisechengyi Looks like I celebrated too early: the error is back again. I cancelled installation of a MacOS VM today because I thought it was solved but I will do that tomorrow to test the "tainted environment" theory.

**Comment by DJFliX** [comment]
I finally got a VM running, installed IDEA + Scala plugin + Pants plugin + JDK 1.8. Cloned the project, ran a test, closed IDEA, attempted to re-run the test and *BAM*. Same error. It almost seems like the pants plugin corrupts something when IDEA shuts down... @wisechengyi Any suggestions on where to look next?

**Comment by mrkb80** [comment]
seeing the exact same issue (as are others).  I have not started looking deeply in to this yet, but maybe I will try to take a pass at it and see what I can find.

**Comment by wisechengyi** [comment]
may have a workaround in #384

**Comment by wisechengyi** [comment]
Fix in release 1.11.1
